lib/include/tree_sitter/api.h

```diff
@@ -86,6 +97,7 @@ typedef struct TSParseState {
 typedef struct TSParseOptions {
   void *payload;
   bool (*interrupt_callback)(TSParseState *state);
+  DecodeFunction decode_callback;
```

Comment:
Why is the `DecodeFunction` included in both `TSInput` and `TSParseOptions`? Could it just be a property of the input?